RuleDocumentation/PossibleIncorrectComparisonWithNull.md

```diff
@@ -7,7 +7,7 @@
 To ensure that PowerShell performs comparisons correctly, the `$null` element should be on the left side of the operator.
 
 There are a number of reasons why this should occur:
-* When there is an array on the left side of a null equality comparison, PowerShell will check for a `$null` IN the array rather than if the array is null.
+* When there is an array on the left side of a null equality comparison, PowerShell will create a new array containing one `$null` item for each `$null` in the original array rather than check if the array is null.
```

Comment:
Can you please provide a reference for the statement *create a new array containing one `$null` item for each `$null` in the original array rather than check if the array is null*?

Reply:
>If there are no matches in a collection, comparison operators do not return anything.

Even that's not completely accurate.  The comparison operator will return an empty array:

```posh
$result = (1..5) -eq $null

$result.GetType().FullName
# System.Object[]

$result.Count
# 0
```

It's a weird gotcha topic, for sure!